### ISSUE TITLE
Execute shell command/script or add node library

### ISSUE THREAD

**Issue by JaniszM** [galenframework/galen#468]
I have quite anyoing problem. I need to send a REST request. With GET there is no problem, I can just use GalenPage. But what with others?

Is there any way to execute system shell command or script and get output of it?
Is there any way to load nodeJS library into Galen tests? Probably I can use load() function to get root .js but this is nasty because requires additional automation to download library directory.

**Comment by JaniszM** [comment]
I think here https://github.com/galenframework/galen/blob/master/galen-core/src/main/java/com/galenframework/javascript/GalenJsApi.java we could add function to execute some shell command. And here https://github.com/galenframework/galen/blob/master/galen-core/src/main/resources/js/GalenApi.js to export it.

Here is nice java code for this:
https://www.mkyong.com/java/how-to-execute-shell-command-from-java/

**Comment by ishubin** [comment]
You can already do it but not in an easy way. From JavaScript you can include apache HttpClient. Here is an example:
```
importPackage(org.apache.http.client.methods);
importClass(org.apache.http.impl.client.DefaultHttpClient);
importClass(org.apache.http.entity.StringEntity);
importClass(org.apache.http.entity.ContentType);
importClass(org.apache.commons.io.IOUtils);

function postJson(url, json) {
    var client = new DefaultHttpClient();
    var post = new HttpPost(url);
    var entity = new StringEntity(JSON.stringify(json), ContentType.create("application/json"));
    post.setEntity(entity);
    var response = client.execute(post);
    var httpResponse = new HttpResponse(response);

    // This is how you could get the response as text
    System.out.println(httpResponse.asText());
    
    return httpResponse;
}
```

I think it would be a good idea to create a simple js wrapper around HttpClient and embed it in galen-core

**Comment by JaniszM** [comment]
Yeah it would be. Since Galen is really great tool for testing, it is missing with a lot of common used tools like rest client. Usually used before and after tests :) Thanks for clue, I will try it as soon as possible.
